collectfast/tests/command/test_copy_hooks.py

```diff
@@ -0,0 +1,117 @@
+from unittest import mock
+from unittest import TestCase
+
+from django.core.files.storage import FileSystemStorage
+from django.test import override_settings as override_django_settings
+
+from collectfast.management.commands.collectstatic import Command
+from collectfast.strategies.base import Strategy
+from collectfast.tests.utils import clean_static_dir
+from collectfast.tests.utils import create_static_file
+from collectfast.tests.utils import make_test
+
+from .test_command import live_test
+from .test_command import make_test_all_backends
+
+
+class BaseTestStrategy(Strategy[FileSystemStorage]):
+    _should_copy_file = None
+
+    def __init__(self, remote_storage):
+        super().__init__(remote_storage)
+        self.post_copy_hook = mock.MagicMock()
+        self.copy_skipped_hook = mock.MagicMock()
+
+    def should_copy_file(
+        self, path, prefixed_path, local_storage
+    ):
+        return self._should_copy_file
+
+
+class TrueTestStrategy(BaseTestStrategy):
+    _should_copy_file = True
+
+
+class FalseTestStrategy(BaseTestStrategy):
+    _should_copy_file = False
+
+
+@make_test
+@override_django_settings(
+    COLLECTFAST_STRATEGY="collectfast.tests.command.test_copy_hooks.TrueTestStrategy",  # noqa: E501
+    STATICFILES_STORAGE="django.core.files.storage.FileSystemStorage",
+)
+def test_calls_post_copy_hook_simple(case: TestCase) -> None:
+    clean_static_dir()
+    path = create_static_file()
+    cmd = Command()
+    cmd.run_from_argv(["manage.py", "collectstatic", "--noinput"])
+    cmd.strategy.post_copy_hook.assert_called_once_with(
+        path.name,
+        path.name,
+        mock.ANY,
```

Comment:
@antonagestam what is the correct storage reference here instead of mock.ANY?

Reply:
Looks like it is set by the finders which I guess could be unpredictable.  I was thinking there would be an easy reference for it.  Since it is typed, probably just leave as mock.ANY

Reply:
Yupp, if you wanted to you could inspect the call arguments closer and check something like `isinstance(storage, Storage)`, but I think this is fine as is.